unstructured/partition/email.py

```diff
@@ -1,26 +1,124 @@
     from typing_extensions import Final
 else:
     from typing import Final
 
-from unstructured.cleaners.core import replace_mime_encodings
-from unstructured.documents.elements import Element, Text
+from unstructured.cleaners.core import replace_mime_encodings, clean_bullets
+from unstructured.cleaners.extract import (
+    extract_ip_address,
+    extract_ip_address_name,
+    extract_mapi_id,
+    extract_datetimetz,
+    extract_email_address,
+)
+from unstructured.documents.email_elements import (
+    EmailElement,
+    BodyText, 
+    Recipient, 
+    Sender, 
+    Subject,
+    ReceivedInfo,
+    MetaData,
+)
+from unstructured.documents.elements import Text, NarrativeText, ListItem, Title
 from unstructured.partition.html import partition_html
+from unstructured.partition.text_type import (
+    is_possible_narrative_text,
+    is_possible_title,
+    is_bulleted_text,
+)
 
 
-VALID_CONTENT_SOURCES: Final[List[str]] = ["text/html"]
+VALID_CONTENT_SOURCES: Final[List[str]] = ["text/html", "text/plain"]
+
+
+def split_by_paragraph(content: str) -> List[str]:
+    return re.split(r"\n\n\n|\n\n|\r\n|\r|\n", content)
+
+
+def parse_received_data(data: str) -> List[EmailElement]:
+
+    ip_address_names = extract_ip_address_name(data)
+    ip_addresses = extract_ip_address(data)
+    mapi_id = extract_mapi_id(data)
+    datetimetz = extract_datetimetz(data)
+
+    elements: List[EmailElement] = list()
+    if ip_address_names and ip_addresses:
+        for name, ip in zip(ip_address_names, ip_addresses):
+            elements.append(MetaData(name=name, text=ip))
+    if mapi_id:
+        elements.append(MetaData(name="mapi_id", text=mapi_id[0]))
+    if datetimetz:
+        elements.append(MetaData(name="received_datetimetz", text=datetimetz[0]))
+
+    return elements
+
+def parse_email_address(data:str)-> Tuple[str, str]:
+    email_address = extract_email_address(data)
+
+    name = re.split("<[a-z0-9\.\-+_]+@[a-z0-9\.\-+_]+\.[a-z]+>"
+    , data.lower())[0].title().strip()
+
+    return name, email_address[0]
+
+def partition_header(msg: List[str])-> List[EmailElement]:
+    elements: List[Text] = list()
+    for item in msg.raw_items():
+        if item[0] == "To":
+            text = parse_email_address(item[1])
+            print(text)
+            elements.append(Recipient(name=text[0], text=text[1]))
+        elif item[0] == "From":
+            text = parse_email_address(item[1])
+            print(text)
+            elements.append(Sender(name=text[0], text=text[1]))
+        elif item[0] == "Subject":
+            print(item[1])
+            elements.append(Subject(text=item[1]))
+        elif item[0] == "Received":
+            print(item[1])
+            elements.append(ReceivedInfo(parse_received_data(item[1])))
+        else:
+            print(item[0], item[1])
+            elements.append(MetaData(name=item[0], text=item[1]))
+
+    return elements
+
+def partition_text(content: List[str]) -> List[EmailElement]:
```

Comment:
`partition_text` I think we should move to a separate `text.py` file under `partition` since we can use that same logic to parse `.txt` files and raw text.